erpnext/public/js/controllers/transaction.js

```diff
@@ -1137,4 +1144,26 @@ erpnext.TransactionController = erpnext.taxes_and_totals.extend({
 
 		return method
 	},
+
+	set_query_for_batch: function(doc, cdt, cdn) {
+		// Show item's batches in the dropdown of batch no
+
+		var me = this;
+		var item = frappe.get_doc(cdt, cdn);
+
+		if(!item.item_code) {
+			frappe.throw(__("Please enter Item Code to get batch no"));
+		} else {
+			filters = {
+				'item_code': item.item_code,
+				'posting_date': me.frm.doc.posting_date || frappe.datetime.nowdate(),
+			}
+			if(item.warehouse) filters["warehouse"] = item.warehouse
+
+			return {
+				query : "erpnext.controllers.queries.get_batch_no",
```

Comment:
@nabinhait,

filtered the batch no based on item_code if doctype is Purchase Receipt